sources/Metal/Renderers/iTermCursorGuideRenderer.m

```diff
@@ -35,6 +38,23 @@ - (void)initializeVerticesWithPool:(iTermMetalBufferPool *)verticesPool {
     self.vertexBuffer = [verticesPool requestBufferFromContext:self.poolContext
                                                      withBytes:vertices
                                                 checkIfChanged:YES];
+
+    const CGRect verticalGuideQuad = CGRectMake(self.margins.left + self.column * cellSize.width,
+                                                self.margins.top,
+                                                cellSize.width,
+                                                cellSize.height * gridSize.height);
+    const iTermVertex verticalGuideVertices[] = {
+        { { CGRectGetMaxX(verticalGuideQuad), CGRectGetMinY(verticalGuideQuad) }, { CGRectGetMaxX(textureFrame), CGRectGetMinY(textureFrame) } },
+        { { CGRectGetMinX(verticalGuideQuad), CGRectGetMinY(verticalGuideQuad) }, { CGRectGetMinX(textureFrame), CGRectGetMinY(textureFrame) } },
+        { { CGRectGetMinX(verticalGuideQuad), CGRectGetMaxY(verticalGuideQuad) }, { CGRectGetMinX(textureFrame), CGRectGetMaxY(textureFrame) } },
+
+        { { CGRectGetMaxX(verticalGuideQuad), CGRectGetMinY(verticalGuideQuad) }, { CGRectGetMaxX(textureFrame), CGRectGetMinY(textureFrame) } },
+        { { CGRectGetMinX(verticalGuideQuad), CGRectGetMaxY(verticalGuideQuad) }, { CGRectGetMinX(textureFrame), CGRectGetMaxY(textureFrame) } },
+        { { CGRectGetMaxX(verticalGuideQuad), CGRectGetMaxY(verticalGuideQuad) }, { CGRectGetMaxX(textureFrame), CGRectGetMaxY(textureFrame) } },
+    };
+    self.verticalGuideVertexBuffer = [verticesPool requestBufferFromContext:self.poolContext
+                                                                  withBytes:verticalGuideVertices
+                                                             checkIfChanged:YES];
```

Comment:
I think you're going to end up needing three vertex buffers:

1. The original one for when there is only a horizontal guide
2. A similar one (but rotated 90°) for when there is only a vertical guide
3. One for the cursor cell for when there are both so the background doesn't get double-drawn.

Reply:
Indeed, but it leads to an asymmetry between the horizontal and vertical guides: the horizontal highlights at the edges of the horizontal texture will be drawn across the cursor while the vertical highlights from the vertical guide (now split in 2) will not.  Aesthetically I'm fine with the asymmetry.  The latest commit of the Metal renderer contains this asymmetry.